Logical model documentation

ArthurMota9 opened the pull request:
### Description

Adds documentation for the following features:
- Constraints;
- Views.

ArthurMota9 committed:
Add videos

ArthurMota9 committed:
Add logical model documentation

ArthurMota9 committed:
Alert that informs feature availability

ArthurMota9 committed:
Add section with conversion to SQL

idmarjr commented on docs/_sidebar.md:
```diff
@@ -15,4 +15,7 @@
+	- [Movendo livremente pelo canvas](canvas-navigation/README.md#movendo-livremente-pelo-canvas)
+- [Modelagem lógica](logical-model/README.md)
+	- [Inclusão de Constraints](logical-model/README.md#inclusão-de-constraints)
+	- [Criação de Views](logical-model/README.md#criação-de-views)
```
Quero sua opinião aqui...  
Pra mantermos um padrão de linguagem, o que tu acha de usarmos sempre o gerúndio

- Incluindo _Constraints_
- Criando _Views_

A gente tem que decidir como tratar esses termos internacionalizados tipo _views_ e _constraints_

idmarjr commented on docs/logical-model/README.md:
```diff
@@ -0,0 +1,72 @@
+- **Default:** É usado para definir um valor padrão para a coluna, caso nenhum valor seja especificado;
+- **Check:** A cláusula _check(p)_ especifica um predicado P que precisa ser satisfeito em cada tupla;
+- **Auto increment:** Permite que um número único seja gerado, automaticamente, quando um novo registro é inserido em uma tabela.
+
```
Duas coisas:

- Eu mudaria isso pra uma tabela
- Precisamos tomar uma decisão sobre como lidar com termos internacionalizados. O que deve e o que não deve ser traduzido.

idmarjr commented on docs/logical-model/README.md:
```diff
@@ -0,0 +1,72 @@
+- **Default:** É usado para definir um valor padrão para a coluna, caso nenhum valor seja especificado;
+- **Check:** A cláusula _check(p)_ especifica um predicado P que precisa ser satisfeito em cada tupla;
+- **Auto increment:** Permite que um número único seja gerado, automaticamente, quando um novo registro é inserido em uma tabela.
+
```
Duas coisas:

- Eu mudaria isso pra uma tabela
- Precisamos tomar uma decisão sobre como lidar com termos internacionalizados. O que deve e o que não deve ser traduzido.

ArthurMota9 replied:
Então, sobre os termos internacionalizados podemos abrir uma discussão @idmarjr @miltonbsn. Acho que em alguns casos é interessante deixar a nomenclatura que é exibida no script SQL, para o usuário se sentir familiarizado.

idmarjr commented on docs/logical-model/README.md:
```diff
@@ -0,0 +1,72 @@
+# Modelagem lógica
+
+Geralmente, essa etapa é resultado da conversão de uma modelagem conceitual, porém, é possível iniciar o projeto de BD, diretamente, pelo projeto lógico, especialmente quando o projetista possui o conhecimento conceitual do projeto.
```
Esse texto tá meio jogado aqui. Minha impressão é que falta uma introdução. Tu começa como se a _conversa_  sobre esse assunto com o leitor já estivesse acontecendo.

idmarjr replied:
BD == ?

ArthurMota9 replied:
Boa! Dei uma introduzida melhor no assunto.

idmarjr commented on docs/logical-model/README.md:
```diff
@@ -0,0 +1,72 @@
+- **Check:** A cláusula _check(p)_ especifica um predicado P que precisa ser satisfeito em cada tupla;
+- **Auto increment:** Permite que um número único seja gerado, automaticamente, quando um novo registro é inserido em uma tabela.
+
+Vale destacar algumas regras que são importantes:
+
+- Não é possível inserir valor **_default_** para colunas que são chave primária ou estrangeira;
+- Chave primária, automaticamente, é **_not null_** e **_unique_**;
+- Não é possível marcar como **_unique_** uma coluna que possui um valor **_default_**;
+- Não é possível marcar como **_auto increment_** uma coluna com valor **_default_** ou chave estrangeira.
```
Esse trecho também merece sua própria _sessão_. Algo como "Regras no uso de constraints"

idmarjr replied:
Uma outra coisa aqui é a forma como tu coloca as regras.
Tu começa com a negação pra só depois explicar onde essa negação acontece.

Eu inverteria essa lógica pra [local] + [regra].  
Algo como:

> - Colunas com o valor **_default_** não podem receber o valor **_unique_**

idmarjr commented on docs/logical-model/README.md:
```diff
@@ -0,0 +1,72 @@
+
+Passos para utilização de _Constraints_:
+
+1. Vá para a tela de listagem de modelos;
+1. Selecione uma modelagem lógica;
```
Esses dois passos pode virar uma introdução:

> Com a modelagem lógica aberta:

1. Selecione a tabela que receberá uma _constraint_'
2. ...

idmarjr commented on docs/logical-model/README.md:
```diff
@@ -0,0 +1,72 @@
+
+Uma visão é uma ferramenta importante no desenvolvimento de banco de dados. Pode ser utilizada para ocultar detalhes e reduzir complexidade do BD, disponibilizar ao usuário apenas o que ele de fato necessita, ser útil como filtro de segurança ou como estratégia de gerenciamento de permissões.
+
+Passos para criação de _Views_:
```
Assim como na linha `28`, acho que que esse pedaço pode ser removido.
Nesse caso em específico esse contexto pode ser incorporado no paragrafo anterior.

idmarjr commented on docs/logical-model/README.md:
```diff
@@ -0,0 +1,72 @@
+	Desculpe, seu navegador não suporta esse formato de vídeo.
+</video>
+
+É possível inserir uma _query expression_ de forma manual por meio do campo de texto, principalmente por usuários experientes com conhecimento em SQL. Para usuários com menos conhecimento é possível clicar no _link_ 'Incluir consulta para criação da visão' que irá abrir um modal que auxiliará na criação da expressão.
```
Esse paragrafo precisa ser retrabalhado.
Tu tá basicamente "desqualificando" o conhecimento do usuário que não sabe  escrever uma _query expression_.

Pensa que esse artigo serve pra ajudar as pessoas a usar a interface do BR Modelo Web. Faz mais sentido você focar em ensinar (Usando o passo-a-passo, como nos outros lugares). Depois disso, insere o vídeo e depois do vídeo põe um paragrafo complementar dizendo que também é possível inserir manualmente. (Que talvez valha um vídeo separado pra deixar bem claro como fazer essa inserção manual)

idmarjr commented on docs/logical-model/README.md:
```diff
@@ -0,0 +1,72 @@
+	Desculpe, seu navegador não suporta esse formato de vídeo.
+</video>
+
+Quando a criação da visão utiliza mais de uma tabela base, é necessário informar os atributos que irão compor o JOIN dessas tabelas.
```
Essa é uma condição bastante especifica. Me pergunto até onde ela deve ser mantida aqui.  
Mas.. se decidirmos que ela deve permanecer aqui, O passo-a-passo tem que estar no corpo do texto também, não só o vídeo.

ArthurMota9 replied:
Boa, é específica mas é importante. Coloquei o passo-a-passo em texto também.

ArthurMota9 committed:
Code review adjustments

idmarjr committed:
WIP: Refinando documentação da modelagem lógica

idmarjr committed:
WIP: Refinando documentação da modelagem lógica

- Movendo a posição da regras no uso de constraits no texto
- Removendo negrito e itálico ao longo do texsto
- Novo text ode introdução para a sessão de views
- Mais detalhes no passo a passo da criação de views
- Mais detalhes no passo a passo da adição de query expressions
- Mais detalhes no passo a passo da da inserção de join em tabelas

ArthurMota9 committed:
Fix typos

idmarjr commented:
@ArthurMota9 Ainda não temos essas alterações disponíveis em produção, certo?

miltonbsn commented:
@ArthurMota9 Poderia seguir com isso aqui? Ja que botamos em produção é importante a doc estar também

idmarjr commented:
> @ArthurMota9 Poderia seguir com isso aqui? Ja que botamos em produção é importante a doc estar também

Me da um dia pra revisar esse MR. Tem alguns ajustes pro conteúdo antes de fazer o merge.

ArthurMota9 commented:
> @ArthurMota9 Poderia seguir com isso aqui? Ja que botamos em produção é importante a doc estar também

Se eu não me engano grande parte dos comentários do @idmarjr foram ajustados por ele nesses commits:
https://github.com/brmodeloweb/brmodelo-docs/pull/1/commits/e38cb8fa30a2c41b76ffe332c6dbf39d0f628bd8 https://github.com/brmodeloweb/brmodelo-docs/pull/1/commits/3ec45c0339023bca1b983018f7a25abba96492a5

Mas vou dar uma revisada nesse PR.

ArthurMota9 committed:
Update README.md

ArthurMota9 commented:
@miltonbsn vou fazer merge desse cara e quando o @idmarjr voltar de férias podemos marcar uma conversa e fazer uns ajustes, mas do jeito que está já ajuda muito 👍🏾

miltonbsn commented:
> @miltonbsn vou fazer merge desse cara e quando o @idmarjr voltar de férias podemos marcar uma conversa e fazer uns ajustes, mas do jeito que está já ajuda muito 👍🏾

Manda ver maninho!!
